### DISCUSSION TITLE
Radio Buttons for Variant Pickers

### DISCUSSION THREAD

**Discussion by useloos** [Shopify/dawn#3290] [Q&A]
Hey! I've been working on my Dawn theme 12.0.0 and I'm having trouble trying to work around with my radio buttons.
I currently have different color pickers for different colors on my products, which are being displayed on the home page of the store, however when I click on one color, let's say off-white, it opens the page of the product variant and I wanted it to just update the current image of the displayed product to match the color picked. 
Any thoughts on how I can follow up with this?

```
{%- if block.settings.picker_type == 'button' -%}
    <variant-radios
      id="variant-radios-{{ section.id }}"
      class="no-js-hidden"
      data-section="{{ section.id }}"
      data-url="{{ product.url }}"
      {% if update_url === false %}
        data-update-url="false"
      {% endif %}
      {{ block.shopify_attributes }}
    >
      {%- for option in product.options_with_values -%}
        <fieldset class="js product-form__input">
          <legend class="form__label">{{ option.name }}</legend>
          {% render 'product-variant-options', product: product, option: option, block: block %}
        </fieldset>
      {%- endfor -%}
      <script type="application/json">
        {{ product.variants | json }}
      </script>
    </variant-radios>
```
The code above is what I've been working on the radio buttons. Even though I have a verification for updating the URL, it does not work. Any thoughts or considerations to it are appreciated.

Thanks.

**Comment by rsusano** [comment]
Hi! For Dawn’s `variant-radios` to update the product media when you select a different variant, the theme JS needs a reliable way to resolve the selected variant’s media.

From your snippet, the important pieces are:
- `data-url="{{ product.url }}"` (Dawn uses this to resolve variant/media)
- `data-update-url="false"` (prevents URL changes, but image updates should still work)
- the `product.variants` JSON (so the JS knows the variants)

What I’d try:
1) Ensure `data-url` is not missing what the script expects. If Dawn resolves with `?variant=...`, set `data-url` to a URL that includes the selected variant query (or check Dawn’s `variant-radios` code to confirm).
2) Confirm the radios generated by `render 'product-variant-options'` produce correct values so selecting a radio actually selects the right variant.
3) Verify your section IDs/classes match what Dawn’s `variant-radios` script targets for the image update.

If you share the rendered HTML of the radio inputs (a small snippet) and whether the media gallery is the standard Dawn one, I can point to the exact attribute/snippet to adjust.

**Comment by meisbellajack** [comment]
If you want to avoid digging deep into global.js (which can break every time you update Dawn), this is exactly why we developed the App Embed approach.

Instead of hacking the theme's radio buttons, you can use **Misk Variant & Product Options.** Because it’s a native App Embed:

**Instant Image Swapping:** It uses AJAX to swap the main product image instantly on the homepage or collection page without a redirect.

**No Code Debt:** It doesn't modify your global.js file, so when you update to Dawn 13 or 14, your variant pickers won't break.

**Unlimited Options:** It also bypasses the native 3-option limit, allowing you to keep your radio buttons for color, but add more complex options (like "Engraving" or "Gift Wrap") that only show up when needed.

It keeps your store looking "great and beautiful" while maintaining the lightning-fast speed you need for the 2026 mobile-first web.
